In [2]:
import pandas as pd
import numpy as np

In [3]:
from pathlib import Path

In [4]:
!pwd

/Users/haohe/Desktop/Heart_shenzhen


In [5]:
root = '/Users/haohe/Desktop/Heart_shenzhen'
audio_data = root + '/audio'
data = root + '/data'

# Process valid_test

In [6]:
df_valid_test = pd.read_excel(data+'/valid_test_mapping.xlsx')
df_valid_test.head()

ComParE2018_Heartbeat.confidential  \
0                          file_name   
1                     devel_0001.wav   
2                     devel_0002.wav   
3                     devel_0003.wav   
4                     devel_0004.wav   

                                          Unnamed: 1  
0                                     orig_file_name  
1  1422900/Ausculation_Report_Eko_Recording_13168...  
2  1423640/Ausculation_Report_Eko_Recording_13169...  
3  633418/Ausculation_Report_Eko_Recording_133855...  
4  1196812/Ausculation_Report_Eko_Recording_14173...

In [7]:
df_valid_test = df_valid_test.rename(columns={
    'ComParE2018_Heartbeat.confidential': 'filename',
    'Unnamed: 1': 'orig_filename'
})

In [8]:
df_valid_test = df_valid_test.iloc[1:,:]
df_valid_test.head()

filename                                      orig_filename
1  devel_0001.wav  1422900/Ausculation_Report_Eko_Recording_13168...
2  devel_0002.wav  1423640/Ausculation_Report_Eko_Recording_13169...
3  devel_0003.wav  633418/Ausculation_Report_Eko_Recording_133855...
4  devel_0004.wav  1196812/Ausculation_Report_Eko_Recording_14173...
5  devel_0005.wav  1109541/Ausculation_Report_Eko_Recording_13871...

In [9]:
df_valid_test['id'] = df_valid_test['orig_filename'].apply(lambda o: int(o.split('/')[0]))
df_valid_test = df_valid_test.sort_values('id')
df_valid_test.head(n=10)

filename                                      orig_filename      id
319  test_0139.wav     11/Ausculation_Report_Eko_Recording_127927.wav      11
307  test_0127.wav     11/Ausculation_Report_Eko_Recording_127928.wav      11
294  test_0114.wav     11/Ausculation_Report_Eko_Recording_127926.wav      11
317  test_0137.wav     11/Ausculation_Report_Eko_Recording_127929.wav      11
229  test_0049.wav  541646/Ausculation_Report_Eko_Recording_139139...  541646
273  test_0093.wav  541646/Ausculation_Report_Eko_Recording_139137...  541646
226  test_0046.wav  541646/Ausculation_Report_Eko_Recording_139138...  541646
201  test_0021.wav  541646/Ausculation_Report_Eko_Recording_139136...  541646
225  test_0045.wav  600789/Ausculation_Report_Eko_Recording_141777...  600789
193  test_0013.wav  600789/Ausculation_Report_Eko_Recording_141778...  600789

In [13]:
df_valid_test['valid'] = df_valid_test['filename'].apply(lambda o: o.split('_')[0])

In [18]:
df_valid_test['valid'] = df_valid_test['valid'].apply(lambda o: True if o == 'devel' else False)

In [28]:
df_valid_test[df_valid_test['valid']==True]['id'].unique()

array([ 633418,  669541,  898855, 1109541, 1192716, 1196812, 1229337,
       1238583, 1246313, 1325598, 1329287, 1338099, 1418854, 1422900,
       1423345, 1423640, 1426007, 1426145, 1426739, 1426840, 1427125,
       1427612, 1427617, 1428510, 1428762, 1429109, 1429473, 1429583,
       1430164, 1430191, 1430374, 1430962, 1431086, 1431674, 1432449])

In [42]:
df_raw = pd.read_csv(data+'/train_label.csv')

In [57]:
df_raw[df_raw['id'] == 1423345]

id name   eng_name gender   cul hometown   age  height  weight  temp  \
77  1423345  李金福  Li, Jinfu    1=男  1=汉族       温州  66.0   160.0    60.0  36.7   

     bpm  tsh  t4  t3  
77  78.0  NaN NaN NaN

In [55]:
unlabel = df_raw[df_raw['tsh'].isnull()]['id'].tolist()

In [56]:
for i in df_valid_test[df_valid_test['valid']==True]['id'].unique().tolist():
    if i in unlabel:
        print(i)

1423345
1426739
1426840
1428510
1428762
1429109
1430374


In [58]:
df_train = pd.read_csv(data+'/train.csv')

In [61]:
for i in df_valid_test[df_valid_test['valid']==False]['id'].unique().tolist():
    if i in unlabel:
        print(i)

11
1369609
1420372
1421960
1424846
1425597
1425630
1427820


In [64]:
df_unlbl = pd.DataFrame(unlabel,columns=['id'])
df_unlbl.head()

id
0       11
1       12
2  1237023
3  1263749
4  1304320

In [65]:
df_unlbl.to_csv(data+'/unlabel.csv',index=False)

In [69]:
df_valid_test = df_valid_test.merge(df_label,how='left',on='id')

In [73]:
df_valid_test.to_csv(data+'/raw_valid_test.csv',index=False)

In [74]:
df_valid_test = df_valid_test[df_valid_test['tsh'].notna()]

In [76]:
df_valid_test.to_csv(data+'/valid_test.csv',index=False)

# Fin

1. train.csv is cleaned train
2. valid_test.csv is cleaned valid / test
3. raw_all.csv is raw all
4. raw_test_valid.csv is raw test valid

In [77]:
df_raw_all = pd.read_csv(data+'/raw_all.csv')
df_raw_all.head()

filename                                      orig_filename      id  \
0  train_0043.wav  524215/Ausculation_Report_Eko_Recording_140111...  524215   
1  train_0209.wav  524215/Ausculation_Report_Eko_Recording_140114...  524215   
2  train_0200.wav  524215/Ausculation_Report_Eko_Recording_140112...  524215   
3  train_0201.wav  524215/Ausculation_Report_Eko_Recording_140113...  524215   
4  train_0159.wav  615897/Ausculation_Report_Eko_Recording_141773...  615897   

  name    eng_name gender   cul hometown   age  height  weight  temp   bpm  \
0  叶思思    Ye, Sisi    1=男  1=汉族       温州  35.0   164.0    64.0  36.5  80.0   
1  叶思思    Ye, Sisi    1=男  1=汉族       温州  35.0   164.0    64.0  36.5  80.0   
2  叶思思    Ye, Sisi    1=男  1=汉族       温州  35.0   164.0    64.0  36.5  80.0   
3  叶思思    Ye, Sisi    1=男  1=汉族       温州  35.0   164.0    64.0  36.5  80.0   
4  叶晓霞  Ye,Xiaoxia    2=女  1=汉族       永嘉  69.0   156.0    50.0  36.9  60.0   

    tsh     t4   t3  
0  1.12   8.32  4.6  
1  1.12   8.32  4.6  
2  1.12   8.32  4.6  
3  1.12   8.32  4.6  
4  2.53  12.34  4.2

In [78]:
df_raw_all[df_raw_all['tsh'].isnull()]

filename                                      orig_filename  \
86   train_0134.wav  1237023/Ausculation_Report_Eko_Recording_13870...   
87   train_0320.wav  1237023/Ausculation_Report_Eko_Recording_13870...   
88   train_0074.wav  1237023/Ausculation_Report_Eko_Recording_13870...   
89   train_0083.wav  1237023/Ausculation_Report_Eko_Recording_13870...   
98   train_0125.wav  1263749/Ausculation_Report_Eko_Recording_14008...   
..              ...                                                ...   
494  train_0146.wav  1432299/SAusculation_Report_Eko_Recording_1417...   
507  train_0076.wav  14326896/Ausculation_Report_Eko_Recording_1410...   
508  train_0140.wav  14326896/Ausculation_Report_Eko_Recording_1410...   
509  train_0324.wav  14326896/Ausculation_Report_Eko_Recording_1410...   
510  train_0259.wav  14326896/Ausculation_Report_Eko_Recording_1410...   

           id name       eng_name gender   cul hometown   age  height  weight  \
86    1237023  林还归   Lin, Huangui    1=男  1=汉族       温州  73.0   170.0    70.0   
87    1237023  林还归   Lin, Huangui    1=男  1=汉族       温州  73.0   170.0    70.0   
88    1237023  林还归   Lin, Huangui    1=男  1=汉族       温州  73.0   170.0    70.0   
89    1237023  林还归   Lin, Huangui    1=男  1=汉族       温州  73.0   170.0    70.0   
98    1263749  陈秋娟  Chen, Qiujuan    2=女  1=汉族       文成  70.0   160.0    77.0   
..        ...  ...            ...    ...   ...      ...   ...     ...     ...   
494   1432299  杨大林     Yang,Dalin    1=男  1=汉族       温州  69.0   168.0    67.0   
507  14326896  NaN       Shi, Xiu    NaN   NaN      NaN   NaN     NaN     NaN   
508  14326896  NaN       Shi, Xiu    NaN   NaN      NaN   NaN     NaN     NaN   
509  14326896  NaN       Shi, Xiu    NaN   NaN      NaN   NaN     NaN     NaN   
510  14326896  NaN       Shi, Xiu    NaN   NaN      NaN   NaN     NaN     NaN   

     temp   bpm  tsh  t4  t3  
86   37.0  73.0  NaN NaN NaN  
87   37.0  73.0  NaN NaN NaN  
88   37.0  73.0  NaN NaN NaN  
89   37.0  73.0  NaN NaN NaN  
98   37.0  82.0  NaN NaN NaN  
..    ...   ...  ...  ..  ..  
494  36.9  88.0  NaN NaN NaN  
507   NaN   NaN  NaN NaN NaN  
508   NaN   NaN  NaN NaN NaN  
509   NaN   NaN  NaN NaN NaN  
510   NaN   NaN  NaN NaN NaN  

[117 rows x 16 columns]

In [19]:
df_raw['orig_filename'][0].split('/')[0]

'1428734'

In [81]:
df_raw['id'] = df_raw['orig_filename'].apply(lambda o: int(o.split('/')[0]))

In [82]:
df_raw = df_raw.sort_values('id')

In [83]:
df_raw.head(n=10)

filename                                      orig_filename      id
42   train_0043.wav  524215/Ausculation_Report_Eko_Recording_140111...  524215
208  train_0209.wav  524215/Ausculation_Report_Eko_Recording_140114...  524215
199  train_0200.wav  524215/Ausculation_Report_Eko_Recording_140112...  524215
200  train_0201.wav  524215/Ausculation_Report_Eko_Recording_140113...  524215
158  train_0159.wav  615897/Ausculation_Report_Eko_Recording_141773...  615897
56   train_0057.wav  615897/Ausculation_Report_Eko_Recording_141775...  615897
288  train_0289.wav  615897/Ausculation_Report_Eko_Recording_141774...  615897
57   train_0058.wav  615897/Ausculation_Report_Eko_Recording_141776...  615897
10   train_0011.wav  648358/Ausculation_Report_Eko_Recording_136364...  648358
111  train_0112.wav  648358/Ausculation_Report_Eko_Recording_136365...  648358

In [38]:
df_label = pd.read_excel(data+'/train_label.xlsx')
df_label.head()

Unnamed: 0 Unnamed: 1   Unnamed: 2 Unnamed: 3 Unnamed: 4 Unnamed: 5  \
0    住院号/门诊号         姓名         英文姓名         性别         民族         籍贯   
1         11         冯建    Feng,Jian        NaN        NaN        NaN   
2         12        NaN  Xu, Jianhua        NaN        NaN        NaN   
3     524215        叶思思     Ye, Sisi        1=男       1=汉族         温州   
4     541646        余贤地  Yu, Xiandi         1=男       1=汉族         文成   

  Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9 Unnamed: 10      Unnamed: 11  \
0      年龄（岁）     身高（cm）     体重（kg）      体温（℃）     心率（bpm）  促甲状腺素TSH（mIU/L）   
1        NaN        NaN        NaN        NaN         NaN              NaN   
2        NaN        NaN        NaN        NaN         NaN              NaN   
3         35        164         64       36.5          80             1.12   
4         62        162         65       36.3          88              0.9   

    Unnamed: 12   Unnamed: 13  
0  游离T4（pmol/L）  游离T3（pmol/L）  
1           NaN           NaN  
2           NaN           NaN  
3          8.32           4.6  
4            11           4.6

In [39]:
df_label = df_label.rename(columns={'Unnamed: 0':'id',
                         'Unnamed: 1':'name',
                         'Unnamed: 2':'eng_name',
                         'Unnamed: 3':'gender',
                         'Unnamed: 4':'cul',
                         'Unnamed: 5':'hometown',
                         'Unnamed: 6':'age',
                         'Unnamed: 7':'height',
                         'Unnamed: 8':'weight',
                         'Unnamed: 9':'temp',
                         'Unnamed: 10':'bpm',
                        'Unnamed: 11':'tsh',
                        'Unnamed: 12':'t4',
                        'Unnamed: 13':'t3'})

In [40]:
df_label = df_label.iloc[1:,:]

In [41]:
df_label.to_csv(data+'/train_label.csv',index=False)

In [48]:
df_raw.to_csv(data+'/train_mapping.csv')

In [54]:
df_label[df_label['id']==1108990]

id name        eng_name gender   cul hometown age height weight  \
23  1108990  陈成绿  Chen, Chenglv     1=男  1=汉族       苍南  74    169     70   

    temp bpm   tsh     t4   t3  
23  36.6  64  0.94  12.23  4.5

In [61]:
df_label[df_label['id'] == 1108990]['tsh'].item()

0.94

In [86]:
df_merge = df_raw.merge(df_label,on='id',how='left')

In [90]:
df_merge.to_csv(data+'/raw_all.csv',index=False)

In [91]:
df_train = df_merge[df_merge['tsh'].notna()]

In [92]:
df_train.to_csv(data+'/train.csv',index=False)

In [93]:
df_train.head()

filename                                      orig_filename      id  \
0  train_0043.wav  524215/Ausculation_Report_Eko_Recording_140111...  524215   
1  train_0209.wav  524215/Ausculation_Report_Eko_Recording_140114...  524215   
2  train_0200.wav  524215/Ausculation_Report_Eko_Recording_140112...  524215   
3  train_0201.wav  524215/Ausculation_Report_Eko_Recording_140113...  524215   
4  train_0159.wav  615897/Ausculation_Report_Eko_Recording_141773...  615897   

  name    eng_name gender   cul hometown age height weight  temp bpm   tsh  \
0  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
1  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
2  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
3  叶思思    Ye, Sisi    1=男  1=汉族       温州  35    164     64  36.5  80  1.12   
4  叶晓霞  Ye,Xiaoxia    2=女  1=汉族       永嘉  69    156     50  36.9  60  2.53   

      t4   t3  
0   8.32  4.6  
1   8.32  4.6  
2   8.32  4.6  
3   8.32  4.6  
4  12.34  4.2

In [95]:
len(df_train.id.unique())

77

In [96]:
df_train.shape

(394, 16)

# Fin

23 of the training data doesn't have label

total 77 unique person and 394 recordings are avaiable